packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3611,6 +3613,62 @@ export default abstract class SqlIntegration
     );
   }
 
+  // Get a Fact Table CTE for segments based on fact tables
+  private getFactSegmentCTE({
```

Comment:
This is very similar to `getFactMetricCTE`, but instead of taking in a metric, it takes in a segment.

I did explore combining these two into a single method `getFactTableCTE` and having it handle either metrics or a segment, but the logic got pretty twisty, and it seems like we handle dates a bit different for `metrics` vs `segments`.

This code definitely doesn't follow the DRY principles, but I don't feel like I have enough context to optimize these functions.